packages/contracts/scripts/deploy/manager/PrintSelectors.ts

```diff
@@ -0,0 +1,44 @@
+import { ethers } from "ethers";
+import path from "path";
+
+const args = process.argv.slice(2);
+
+if (args.length != 1) {
+  console.log(`please supply the correct parameters:
+    facetName
+  `);
+  process.exit(1);
+}
+
+async function printSelectors(contractName: string, artifactFolderPath = "../../../out") {
+  const contractFilePath = path.join(
+    artifactFolderPath,
+    `${contractName}.sol`,
+    `${contractName}.json`
+  );
+  const contractArtifact = require(contractFilePath);
+  const abi = contractArtifact.abi;
+  const bytecode = contractArtifact.bytecode;
+  const target = new ethers.ContractFactory(abi, bytecode);
+  const signatures = Object.keys(target.interface.functions);
+
+  const selectors = signatures.reduce((acc, val) => {
+    if (val !== "init(bytes)") {
+      // @ts-ignore
+      acc.push(target.interface.getSighash(val));
```

Comment:
Is this going to cause issues? I don't know Foundry or Solidity well.

Reply:
> Is this going to cause issues? I don't know Foundry or Solidity well.

In diamond, function with the same name can exist only one, even they are on difference facets
And it's important to check if there is any duplicated function before adding a new facet or new functions
PrintSelectors.ts file is used to print the function selectors of any facet